Roles/Impostor/Kamikaze.cs

```diff
@@ -0,0 +1,130 @@
+{
+    private static readonly int Id = 26900;
+    public static List<byte> MaxMarked = new();
+    public static bool IsEnable = false;
+
+
+    public static Dictionary<byte, byte> KamikazedList = new();
+    private static OptionItem KillCooldown;
+    private static OptionItem OptMaxMarked;
+    public static Dictionary<byte, int> MarkedLim = new();
+
+    public static bool CheckKamiDeath = false;
+
+
+
+    public static void SetupCustomOption()
+    {
+        SetupRoleOptions(Id, TabGroup.ImpostorRoles, CustomRoles.Kamikaze);
+        KillCooldown = FloatOptionItem.Create(Id + 10, "KillCooldown", new(0f, 180f, 2.5f), 25f, TabGroup.ImpostorRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.Kamikaze])
+            .SetValueFormat(OptionFormat.Seconds);
+        OptMaxMarked = IntegerOptionItem.Create(Id + 11, "KamikazeMaxMarked", new(1, 14, 1), 14, TabGroup.ImpostorRoles, false).SetParent(Options.CustomRoleSpawnChances[CustomRoles.Kamikaze])
+           .SetValueFormat(OptionFormat.Times);
+
+    }
+    public static void Init()
+    {
+        MaxMarked = new();
+        MarkedLim = new();
+        KamikazedList = new();
+        IsEnable = false;
+    }
+    public static void Add(byte playerId)
+    {
+        IsEnable = true;
+        MarkedLim.Add(playerId, OptMaxMarked.GetInt());
+
+        // Double Trigger
+        var pc = Utils.GetPlayerById(playerId);
+        pc.AddDoubleTrigger();
+
+        if (!AmongUsClient.Instance.AmHost) return;
+        if (!Main.ResetCamPlayerList.Contains(playerId))
+            Main.ResetCamPlayerList.Add(playerId);
+    }
+
+    private static void SendRPC(byte KamiId, byte targetId, byte KillerIdo)
+    {
+        MessageWriter writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId, (byte)CustomRPC.SyncKami, SendOption.Reliable, -1);
+        writer.Write(KamiId);
+        writer.Write(targetId);
+        writer.Write(MarkedLim[KillerIdo]);
+        AmongUsClient.Instance.FinishRpcImmediately(writer);
+    }
+    public static void ReceiveRPC(MessageReader reader)
+    {
+        
+        var KamiId = reader.ReadByte();
+        var targetId = reader.ReadByte();
+        byte KillerIdo = reader.ReadByte();
+        int Limit = reader.ReadInt32();
+        
+        KamikazedList[targetId] = KamiId;
+        MarkedLim.Add(KillerIdo, Limit);
+        MarkedLim[KillerIdo] = Limit;
+    }
+    public static void MurderKamikazedPlayers(PlayerControl kamikameha)
+    {
+        if (!KamikazedList.ContainsKey(kamikameha.PlayerId)) return;
+
+
+        if (!kamikameha.IsAlive())
+        {
+            KamikazedList.Remove(kamikameha.PlayerId);
+        }
+        else if(CheckKamiDeath)
+        {
+            foreach (var alivePlayer in Main.AllAlivePlayerControls)
+            {
+                Main.PlayerStates[kamikameha.PlayerId].deathReason = PlayerState.DeathReason.Targeted;
+                alivePlayer.RpcMurderPlayerV3(kamikameha);
+            }
```

Comment:
Why do all alive players murder this one single player?!

Reply:
OH RIGHT, I forgot to check if aliveplayer corresponds with kamikameha player. Fixed now.